datafusion/core/tests/custom_sources_cases/provider_filter_pushdown.rs

```diff
@@ -134,9 +134,19 @@ impl ExecutionPlan for CustomPlan {
         _partition: usize,
         _context: Arc<TaskContext>,
     ) -> Result<SendableRecordBatchStream> {
+        let schema_captured = self.schema().clone();
         Ok(Box::pin(RecordBatchStreamAdapter::new(
             self.schema(),
-            futures::stream::iter(self.batches.clone().into_iter().map(Ok)),
+            futures::stream::iter(self.batches.clone().into_iter().map(move |batch| {
```

Comment:
Please see https://github.com/apache/datafusion/issues/18782#issuecomment-3568216125 for my thoughts/reason on updating this test.